vm/actor/tests/paych_actor_test.rs

```diff
@@ -0,0 +1,1192 @@
+    let (mut rt, mut sv) = require_create_cannel_with_lanes(&bs, 1);
+    let state: PState = rt.get_state().unwrap();
+    let other_addr = Address::new_id(104);
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+    sv.extra = Some(ModVerifyParams {
+        actor: other_addr,
+        method: Method::UpdateChannelState as u64,
+        data: Serialized::serialize([1, 2, 3, 4]).unwrap(),
+    });
+    rt.expect_validate_caller_addr(&[state.from, state.to]);
+    let exp_send_params = PaymentVerifyParams {
+        extra: Serialized::serialize(vec![1, 2, 3, 4]).unwrap(),
+        proof: vec![],
+    };
+    rt.expect_send(
+        other_addr,
+        Method::UpdateChannelState as u64,
+        Serialized::serialize(exp_send_params).unwrap(),
+        TokenAmount::from(0u8),
+        Serialized::default(),
+        ExitCode::ErrPlaceholder,
+    );
+    rt.expect_verify_signature(
+        sv.clone().signature.unwrap(),
+        state.to,
+        to_vec(&sv).unwrap(),
+        ExitCode::Ok,
+    );
+    let ucp = UpdateChannelStateParams {
+        proof: vec![],
+        secret: vec![],
+        sv: sv.clone(),
+    };
+    let v = rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::UpdateChannelState as u64,
+            &Serialized::serialize(ucp).unwrap(),
+        )
+        .unwrap_err();
+    assert_eq!(v.exit_code(), ExitCode::ErrPlaceholder);
+    rt.verify();
+}
+
+#[test]
+fn update_channel_setting() {
+    let bs = MemoryDB::default();
+    let (mut rt, sv) = require_create_cannel_with_lanes(&bs, 1);
+    rt.epoch = 10;
+    let state: PState = rt.get_state().unwrap();
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), state.from);
+    rt.expect_validate_caller_addr(&[state.from, state.to]);
+    assert!(rt
+        .call(
+            &PAYCH_ACTOR_CODE_ID.clone(),
+            Method::Settle as u64,
+            &Serialized::default()
+        )
+        .is_ok());
+
+    let exp_settling_at = SETTLE_DELAY + 10;
+    let state: PState = rt.get_state().unwrap();
+    assert_eq!(exp_settling_at, state.settling_at);
+    assert_eq!(state.min_settle_height, 0);
+
+    let min_settle_vec = vec![0, 2, 12];
+    let exp_min_settle_height = vec![state.min_settle_height, 2, 12];
+    let exp_settling_at = vec![state.settling_at, state.settling_at, 12];
+    let num_test_cases = min_settle_vec.len();
+
+    for i in 0..num_test_cases {
+        let mut ucp = UpdateChannelStateParams {
+            proof: vec![],
+            secret: vec![],
+            sv: sv.clone(),
+        };
+        ucp.sv.min_settle_height = exp_min_settle_height[i];
+        rt.expect_validate_caller_addr(&[state.from, state.to]);
+        rt.expect_verify_signature(
+            ucp.sv.clone().signature.unwrap(),
+            state.to,
+            to_vec(&ucp.sv).unwrap(),
+            ExitCode::Ok,
+        );
+        assert!(rt
+            .call(
+                &PAYCH_ACTOR_CODE_ID.clone(),
```

Comment:
```suggestion
                &*PAYCH_ACTOR_CODE_ID,
```
I didn't verify in code, but you should be able to do this for these

Reply:
Yup changed added